### ISSUE TITLE
*time.Time decoding fail with "invalid code=xx decoding bytes length"

### ISSUE THREAD

**Issue by jibidus** [vmihailenco/msgpack#256]
I got an error during encoding/decoding time pointer with such code:

```go
func (t *MsgpackTest) TestTimePtr(c *C) {
	in := time.Date(2009, 11, 17, 20, 34, 58, 651387000, time.UTC)
	var out *time.Time
	c.Assert(t.enc.Encode(in), IsNil)
	c.Assert(t.dec.Decode(&out), IsNil)
}
```

The error:
```
Expected :
Actual   :*errors.errorString = &errors.errorString{s:"msgpack: invalid code=d7 decoding bytes length"} ("msgpack: invalid code=d7 decoding bytes length")
```

With msgpack version *4.3.10*, Go version 1.13.4, on macOS, local FR.

**Comment by vmihailenco** [comment]
Thanks for the report - should be fixed in v4.3.11

**Comment by abvarun226** [comment]
Note for others on v4: the fix is in v4.3.11 but didn't make it to v4.3.12

https://github.com/vmihailenco/msgpack/blob/v4.3.11/decode_value.go#L65-L69
https://github.com/vmihailenco/msgpack/blob/v4.3.12/decode_value.go#L64

**Comment by ripexz** [comment]
> Note for others on v4: the fix is in v4.3.11 but didn't make it to v4.3.12
> 
> https://github.com/vmihailenco/msgpack/blob/v4.3.11/decode_value.go#L65-L69 https://github.com/vmihailenco/msgpack/blob/v4.3.12/decode_value.go#L64

That's caught us out and we're not quite ready migrate up a major version... @vmihailenco any chance you could bring v4.3.11 back into v4 branch (https://github.com/vmihailenco/msgpack/compare/v4...v4.3.11) and release v4.3.13? 😊
